app/controllers/api/v1/exercises_controller.rb

```diff
@@ -132,6 +132,29 @@ def update
       standard_update(@exercise, nil, user: current_api_user)
     end
 
+
+
+    ##################################################################
+    ## post save_image                                              ##
+    ##################################################################
+
+    api :POST, '/exercises/:uid/save_image', 'Save an image onto an exercise'
+    description <<-EOS
+      Saves an image asset as an attachment on an Exercise.
+
+      Unlike other API calls, this is accomplished via a multi-part form upload
+      with a file part, not as a traditional POST of JSON data
+
+      Requires a single form parameter named "image"
+
+      #{json_schema(Api::V1::AttachmentRepresenter, include: :readable)}
+    EOS
+    def save_image
+      attachment = AttachFile.call(@exercise, params[:image].tempfile).outputs[:attachment]
```

Comment:
Actually the access check is present on the `get_exercise` method, which is called as a before_filter.  

If the user doesn't have access it renders 404.  I think that's what we want because if we return `access_denied` then we're revealing the presence of an exercise because others would return 404.  

It's all moot though since I'm moving logic to an `attachments` controller :grin: